src/utils/getTokenBalance.js

```diff
@@ -0,0 +1,90 @@
+require("dotenv").config('../../.env');
+const {
+    tokens,
+    erc20MinABI
+} = require('./constants.js')
+const uploadS3 = require('./uploadS3');
+const dateUtils = require('./dateUtils');
+
+const Web3 = require('web3');
+const rpcURL = process.env.MAIN_RPC_URL
+// Show Web3 where it needs to look for a connection to Ethereum.
+web3 = new Web3(new Web3.providers.HttpProvider(rpcURL));
+
+// let tokenAddress = USDT_TOKEN_ADDRESS
+let walletAddress = process.env.BITDAO_ACCOUNT
+
+/**
+ * Get erc20 token balance.
+ * @param tokenAddress erc20 token address
+ * @returns balance of erc20
+ */
+async function getBalance(tokenAddress) {
+    let contract = new web3.eth.Contract(erc20MinABI, tokenAddress);
+    decimals = await contract.methods.decimals().call();
+    balance = await contract.methods.balanceOf(walletAddress).call();
+    balance = balance / (Math.pow(10, decimals))
+    return balance;
+}
+
+/**
+ * Get ETH balance of account.
+ * @returns 
+ */
+async function getEthBalance() {
+    const etherBalance = await web3.eth.getBalance(walletAddress);
+    return web3.utils.fromWei(etherBalance, 'ether');
+}
+
+async function getTokensBalance() {
+    let errorOccur
+    for (var i = 0; i < tokens.length; i++) {
```

Comment:
Fairly nitpick-y, but I think this `for` body can be a lot cleaner by abstracting the logic S3-related logic to only be written once:

```
            var getter = getEthBalance;

            if (tokens[i].symbol !== 'ETH') {
                getter = () => getBalance(tokens[i].tokenAddress);
            }

            await getter().then(function (balance) {
                const fileName = getFileName(tokens[i].symbol)
                const content = getContent(tokens[i].symbol, balance);
                uploadS3.uploadFile(fileName, JSON.stringify(content));
            }).catch((err) => {
                console.log('get ether balance error:', err);
                errorOccur = true
            })
```